sencha-workspace/SlateTasksTeacher/app/controller/Dashboard.js

```diff
@@ -458,8 +468,276 @@ Ext.define('SlateTasksTeacher.controller.Dashboard', {
         });
     },
 
+    onAddGoogleAttachmentClick: function() {
+        var me = this,
+            taskEditor = me.getTaskEditor(),
+            token = Ext.util.Cookies.get('googleAppsToken', '/');
+
+        if (!taskEditor.getTask().phantom) {
+            Ext.Msg.alert('Error', 'Unable to add attachments while editing&hellip; Please create a new task.');
+            return;
+        }
+        // check for google token in cookies
+        gapi.load('auth:picker:client', {
+            callback: function() {
+                // TODO: check if token is any good
+                if (token) {
+                    return me.doOpenFilePicker(token);
+                }
+
+                me.doGoogleAuthentication();
+            }
+        });
+    },
+
+    onGoogleAuthentication: function(authResult) {
+        var me = this,
+            googleAppsDomain = SiteEnvironment && SiteEnvironment.googleAppsDomain || location.host,
+            googleAppsEmailRegex = new RegExp('^.+\@'+googleAppsDomain+'$'),
+            tokenExpiration,
+            _verifyUserIsWithinDomain, _userIsNotWithinDomain;
+
+        console.info('handleAuthResult(%o)', authResult);
+
+        if (!authResult || authResult.error) {
+            Ext.Msg.alert('Error', 'Unable to authenticate user. Please try again or contact an administrator.');
+            return;
+        }
+
+        _userIsNotWithinDomain = function(errorMessage) {
+            Ext.Msg.alert('Error', errorMessage);
+            return;
+        };
+
+        _verifyUserIsWithinDomain = function(response) {
+            console.info('loaded user', response.result);
+
+            if (!response.result || !response.result.user || !response.result.user.emailAddress.match(googleAppsEmailRegex)) {
+                return _userIsNotWithinDomain('Account must be under the correct domain. ' + googleAppsDomain);
```

Comment:
@nbey after this dialog is shown, clicking "attachment" again jumps right back to it without letting the user sign in again. See if there is a client api call you can make to "deauth" the user from the current js api session, ideally without signing them out of google. If you reload the page you can pick an account again so there should be a call in the js API to achieve the same thing